src/bin/nydus-image/deduplicate.rs

```diff
@@ -0,0 +1,361 @@
+// 2022 Nydus Developers.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+//! Deduplicate for Chunk
+
+use anyhow::{Context, Result};
+use nydus_api::ConfigV2;
+use nydus_builder::Tree;
+use nydus_rafs::metadata::RafsSuper;
+use nydus_storage::device::BlobInfo;
+use rusqlite::{params, Connection};
+use std::error::Error;
+use std::fs;
+use std::path::Path;
+use std::sync::{Arc, Mutex};
+pub trait Database {
+    fn create_chunk(&self) -> Result<(), Box<dyn Error + Send + Sync>>;
+    fn create_blob(&self) -> Result<(), Box<dyn Error + Send + Sync>>;
+    fn insert_chunk(&self, chunk: &ChunkTable) -> Result<(), Box<dyn Error + Send + Sync>>;
+    fn insert_blob(&self, blob: &BlobTable) -> Result<(), Box<dyn Error + Send + Sync>>;
+    fn get_chunks(&self) -> Result<Vec<ChunkTable>, Box<dyn Error + Send + Sync>>;
+    fn get_blobs(&self) -> Result<Vec<BlobTable>, Box<dyn Error + Send + Sync>>;
+}
+
+pub struct SqliteDatabase {
+    conn: Arc<Mutex<Connection>>,
+}
+
+impl SqliteDatabase {
+    pub fn new(database_path: &str) -> Result<Self, rusqlite::Error> {
+        // Delete the database file if it exists
+        if let Ok(metadata) = fs::metadata(database_path) {
+            if metadata.is_file() {
+                if let Err(err) = fs::remove_file(database_path) {
+                    eprintln!(
+                        "Warning: Unable to delete existing database file: {:?}",
+                        err
+                    );
+                }
+            }
+        }
+
+        // Attempt to open a new SQLite connection
+        let conn = Connection::open(database_path)?;
+
+        Ok(Self {
+            conn: Arc::new(Mutex::new(conn)),
+        })
+    }
+}
+
+impl Database for SqliteDatabase {
+    fn create_chunk(&self) -> Result<(), Box<dyn Error + Send + Sync>> {
```

Comment:
Can we use `anyhow::Error` as the default Error type?

Reply:
Already modified.